packages/fast-components-styles-msft/src/button/index.ts

```diff
@@ -1,16 +1,186 @@
+        borderRadius: toPx(2),
+        cursor: "pointer",
+        overflow: "hidden",
+        lineHeight: "1",
+        textAlign: "center",
+        whiteSpace: "nowrap",
+        verticalAlign: "bottom",
+        transition: "all 0.2s ease-in-out",
         color: (config: IDesignSystem): string => {
-            return config.foregroundColor;
+            return config.backgroundColor;
+        },
+        ...applyMixedColor("backgroundColor", 0.46),
+        "&:hover": {
+            ...applyMixedColor("backgroundColor", 0.46, 0.8)
+        },
+        "&:focus": {
+            outline: "none",
+            ...applyMixedColor("backgroundColor", 0.38)
+        },
+        "&:disabled, &[aria-disabled]": {
+            opacity: ".4",
+            cursor: "not-allowed",
+            "&:hover": {
+                ...applyMixedColor("backgroundColor", 0.46)
+            }
+        }
+    },
+    button_primary: {
+        extend: "button",
+        color: (config: IDesignSystem): string => {
+            return config.backgroundColor;
         },
         backgroundColor: (config: IDesignSystem): string => {
             return config.brandColor;
+        },
+        "&:hover": {
+            backgroundColor: (config: IDesignSystem): string => {
+                return Chroma(config.brandColor).alpha(0.8).css();
+            }
+        },
+        "&:focus": {
+            outline: "none",
+            backgroundColor: (config: IDesignSystem): string => {
+                return Chroma(config.brandColor).darken().css();
+            }
+        },
+        "&:disabled, &[aria-disabled]": {
+            backgroundColor: (config: IDesignSystem): string => {
+                return config.brandColor;
+            },
+            "&:hover": {
+                backgroundColor: (config: IDesignSystem): string => {
+                    return config.brandColor;
+                }
+            }
         }
     },
+    button_outline: {
+        extend: "button",
+        color: (config: IDesignSystem): string => {
+            return config.foregroundColor;
+        },
+        ...applyMixedColor("borderColor", 0.46),
+        ...applyTransaprentBackground(),
+        "&:hover": {
+            ...applyMixedColor("borderColor", 0.46, 0.8),
+            ...applyTransaprentBackground()
+        },
+        "&:focus": {
+            borderColor: "transparent",
+            boxShadow: `0 0 0 ${toPx(2)}`,
+            ...applyMixedColor("boxShadowColor", 0.46),
+            ...applyTransaprentBackground()
+        },
+        "&:disabled, &[aria-disabled]": {
+            ...applyTransaprentBackground(),
+            "&:hover": {
+                ...applyTransaprentBackground()
+            }
+        }
+    },
+    button_lightweight: {
+        ...applyTransaprentBackplateStyles()
+    },
+    button_justified: {
+        ...applyTransaprentBackplateStyles(),
+        padding: `${toPx(23)} ${toPx(2)} ${toPx(2)} 0`

```

Comment:
there is now a location utility @nicholasrice created that can be used here

Reply:
Cool - updating